tests/src/Unit/Plugin/Tamper/ArrayFilterTest.php

```diff
@@ -11,13 +11,34 @@
 class ArrayFilterTest extends UnitTestCase {
 
   /**
-   * Test the plugin.
+   * @var \Drupal\tamper\Plugin\Tamper\ArrayFilter
    */
-  public function testArrayFilter() {
-    $config = [];
-    $plugin = new ArrayFilter($config, 'array_filter', []);
-    $tampered = $plugin->tamper(['foo', '', 'bar', FALSE, 'baz', [], 'zip']);
-    $this->assertArrayEquals(['foo', 'bar', 'baz', 'zip'], $tampered);
+  protected $plugin;
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function setUp() {
+    $this->plugin = new ArrayFilter([], 'array_filter', []);
+    parent::setUp();
+  }
+
+  /**
+   * Test the array filter plugin with a single value.
+   */
+  public function testArrayFilterWithSingleValue() {
+    $original = 'foo';
+    $expected = ['foo'];
```

Comment:
This is probably actually crazy right? On the other hand, if we applying this plugin we probably expect an array.

Reply:
Migrate throws an exception if you supply a scalar value to a plugin that deals with arrays. See for example the 'extract' process plugin. We could decide to do nothing in this case or throw an exception.

Reply:
Sounds good - I like the idea of throwing an exception if the type isn't expected, assuming that if the exception is thrown then feeds tamper can make the decision on using the original value?